packages/node_modules/node-red/settings.js

```diff
@@ -483,4 +484,53 @@ module.exports = {
     //    *   - reason: if result is false, the HTTP reason string to return
     //    */
     //},
+
+    subflowModule: {
+        /** Whether or not to allow downloading NPM packaged subflow.
+         ** default is true.
+         */
+        allowDownload: true,
+        /** When creating NPM packaged subflow the following directory
+         ** is looked up for templates.
+         */
+        //templateDir: '/home/nol/subflow-templates'
+        /** Whether or not to allow encoding of subflow JSON file within
+         ** NPM packaged subflow. default is true.
+         */
+        allowEncoding: true,
+    },
+    /** Specification of additional encoding methods
+     */
+    encodeSubflow: {
```

Comment:
See my main comments, but I do not think this setting should be included in the default settings file. The documentation for working with subflows should be updated to include examples of this setting.

If this design used plugins, there would be no need for this setting to even exist.

If it does exist, it should exist under the `subflowModule` setting - not as a separate top-level setting.

Reply:
In this update, moved it to `subflowModule` setting.  
If plugin mechanism is used, I will consider removing them.